src/pallene/uninitialized.lua

```diff
@@ -3,149 +3,102 @@
 -- Please refer to the LICENSE and AUTHORS files for details
 -- SPDX-License-Identifier: MIT
 
+-- In this module we detect when variables are used before being initialized and when control flows
+-- to the end of a non-void function without returning. The analysis is fundamentally a dataflow
+-- one, but we don't check for convergence (this set of properties seem to converge in one pass) and
+-- we don't merge the properties of one basic block with one that it jumps to if the "jumped to"
+-- block has a lower id, that is, we don't consider backwards jumps (that's how we deal with loops
+-- for now).  Make sure that you call ir.clean first, so that it does the right thing in the
+-- presence of `while true` loops.
```

Comment:
this comment needs to be updated (e.g. now we do check for convergence and the dataflow is as explicit as it gets)

Reply:
You're right. I'll update it.